src/pods/canvas/canvas.business.ts

```diff
@@ -73,3 +79,75 @@ export const calculateRelationXCoordinate = (
   xOrigin: calculateRelationXCoordinateOrigin(tableOrigin, tableDestination),
   xDestination: calculateRelationXCoordinateEnd(tableOrigin, tableDestination),
 });
+
+const seekField = (
+  fieldId: GUID,
+  YPosition: number,
+  fields: FieldVm[]
+): seekResult => {
+  let found = false;
+  let parentCollapsed = false;
+  let newYPosition = YPosition;
+
+  for (let i = 0; i < fields.length && !found; i++) {
+    const field = fields[i];
+    if (field.id === fieldId) {
+      found = true;
+      return {
+        found,
+        parentCollapsed,
+        YPosition: newYPosition,
+      };
+    } else {
+      if (!parentCollapsed) {
+        newYPosition += ROW_HEIGHT;
+      }
+
+      if (!parentCollapsed && field.isCollapsed) {
+        parentCollapsed = true;
+      }
+
+      if (
+        field.type === 'object' &&
+        field.children &&
+        field.children.length > 0
+      ) {
```

Comment:
This _if_ statement is complex, we can wrap it into a helper function, something like:

```ts
const doesFieldcontainsChildren = (field: Field) => (field.type === 'object' && field.children &&  field.children.length > 0); 
```

Reply:
It's fixed